lib/clients/reports.ex

```diff
@@ -21,9 +25,9 @@ defmodule GlassFactoryApi.Clients.Reports do
         }
       ]
   """
-  @spec list_time_reports(integer(), map()) :: {:error, String.t()} | {:ok, [TimeReport.t()]}
-  def list_time_reports(client_id, config \\ %{}) do
-    client_request = ApiClient.get("clients/#{client_id}/reports/time", [], config)
+  @spec list_time_reports(integer, Keyword.t, map) :: {:error, String.t} | {:ok, [TimeReport.t]}
```

Comment:
We can create a type to specify which keys are allowed and use it here to replace `Keyword.t`.

Reply:
Just wondering, we should have more endpoints with parameters that will generate more types.  Is it ok to generate a type for every different set of parameters we need?

Reply:
It is useful for documentation purposes. Especially now that their API is still under heavy development and things are inconsistent, take the `start` vs. `start_date` issue for example.